### PR TITLE
chore(suite): add SettingsSectionItem

### PR DESCRIPTION
**Pull request by komret**
Repetition is the mother of wisdom, they say. Or is it?

Just refactoring, no functional changes.

### CODE REVIEW

File: packages/suite/src/components/settings/SettingsSectionItem.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { useAnchor } from 'src/hooks/suite/useAnchor';
+import { SettingsAnchor } from 'src/constants/suite/anchors';
+
+interface SettingsSectionItemProps {
+    anchorId: SettingsAnchor;
+    children: ReactNode;
+}
+
+export const SettingsSectionItem = ({ anchorId, children }: SettingsSectionItemProps) => {
+    const { anchorRef, shouldHighlight } = useAnchor(SettingsAnchor.PinProtection);
```
Comment (tomasklim) [inline packages/suite/src/components/settings/SettingsSectionItem.tsx]: you probably wanted to use anchorId here

File: packages/suite-web/e2e/tests/settings/t2t1-device-settings.test.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ describe('T2T1 - Device settings', () => {
         cy.passThroughInitialRun();
 
         cy.log('Try to change device homescreen');
-        cy.getTestElement('@settings/device/homescreen').scrollIntoView();
+        cy.getTestElement('@device-settings/homescreen').scrollIntoView();
```
Comment (tomasklim) [inline packages/suite-web/e2e/tests/settings/t2t1-device-settings.test.ts:86]: you can import anchor constants here and use it

**Review by tomasklim — CHANGES_REQUESTED** [review]

File: packages/suite/src/components/settings/SettingsSectionItem.tsx (hunk 1 above)
Comment (komret) [reply packages/suite/src/components/settings/SettingsSectionItem.tsx]: 239dd1703997060338d34ea7ae49b149e94a033c

File: packages/suite-web/e2e/tests/settings/t2t1-device-settings.test.ts (hunk 2 above)
Comment (komret) [reply packages/suite-web/e2e/tests/settings/t2t1-device-settings.test.ts:86]: 239dd1703997060338d34ea7ae49b149e94a033c

Comment (komret) [reply packages/suite-web/e2e/tests/settings/t2t1-device-settings.test.ts:86]: I had to revert it.

Screenshot 2024-04-17 at 10 59 34

**Review by tomasklim — APPROVED** [review]

**Pull request merged**